### PR TITLE
refactor: fix type inconsistency

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #10~~

**Description**
- fix type inconsistency

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by michalsn** [comment]
Good catch. Thanks.

File: src/Commands/QueueWork.php [hunk 1]
```diff
@@ -93,8 +93,8 @@ public function run(array $params)
         $maxTime    = $params['max-time'] ?? CLI::getOption('max-time') ?? 0;
         $memory     = $params['memory'] ?? CLI::getOption('memory') ?? 128;
         $priority   = $params['priority'] ?? CLI::getOption('priority') ?? $config->getQueuePriorities($queue) ?? 'default';
-        $tries      = $params['tries'] ?? CLI::getOption('tries');
-        $retryAfter = $params['retry-after'] ?? CLI::getOption('retry-after');
+        $tries      = (int) ($params['tries'] ?? CLI::getOption('tries'));
+        $retryAfter = (int) ($params['retry-after'] ?? CLI::getOption('retry-after'));
```
Comment (michalsn) [inline src/Commands/QueueWork.php]: This has to be handled differently. We expect `null` if no value is set.

Comment (kenjis) [reply src/Commands/QueueWork.php]: Oh, thanks. Fixed. https://github.com/codeigniter4/queue/pull/11/commits/8ec75d7cb1f767475e95ca84c0d025d68888d359#diff-29248c72207c00e6d489cddcd4e5fa9f33026a5f8503ea0b6b4f73401a3e167c

Comment (kenjis) [reply src/Commands/QueueWork.php]: And should be: 7cafc2d0bb56fe0e2223505a8727211be82351fd

**Comment by kenjis** [comment]
It seems the remaining errors are bugs or API inconsistency.

```
 ------ ----------------------------------------------------------------------------------------- 
  Line   src/Commands/QueueWork.php                                                               
 ------ ----------------------------------------------------------------------------------------- 
  195    Call to an undefined method CodeIgniter\Queue\Interfaces\JobInterface::getTries().       
  197    Call to an undefined method CodeIgniter\Queue\Interfaces\JobInterface::getRetryAfter().  
 ------ ----------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------- 
  Line   src/Handlers/BaseHandler.php                                                               
 ------ ------------------------------------------------------------------------------------------- 
  59     Call to an undefined method CodeIgniter\Queue\Handlers\BaseHandler::push().                
  72     Call to an undefined method CodeIgniter\Queue\Models\QueueJobFailedModel::affectedRows().  
  84     Call to an undefined method CodeIgniter\Queue\Models\QueueJobFailedModel::truncate().      
 ------ ------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------------------------- 
  Line   src/Handlers/PredisHandler.php                                                           
 ------ ----------------------------------------------------------------------------------------- 
  97     Parameter #2 $fields of method Predis\ClientInterface::hdel() expects array, int given.  
  112    Parameter #2 $fields of method Predis\ClientInterface::hdel() expects array, int given.  
  125    Parameter #2 $fields of method Predis\ClientInterface::hdel() expects array, int given.  
 ------ ----------------------------------------------------------------------------------------- 
```

**Comment by michalsn** [comment]
Thanks! 

Seems like  `QueueJobFailedModel::affectedRows()` and `QueueJobFailedModel::truncate()` are fine, the rest is fixed in #12.

File: src/Commands/QueueWork.php (hunk 1 above)
Comment (michalsn) [reply src/Commands/QueueWork.php]: Thanks. I made changes to support all the params: https://github.com/codeigniter4/queue/pull/12/commits/440d844007bc3e62fddead9f710a6968b30afbd0

**Comment by kenjis** [comment]
I don't understand the code well, but is that model unrelated to Redis?

And you created #12, so can I close this PR?

**Comment by michalsn** [comment]
> I don't understand the code well, but is that model unrelated to Redis?

No. Failed jobs are always stored in the "traditional" database.

> And you created https://github.com/codeigniter4/queue/pull/12, so can I close this PR?

Yes, however, please don't remove the branch yet - just in case.

**Pull request merged**